pytorch_lightning/loops/base.py

```diff
@@ -85,14 +110,13 @@ def run(self, *args: Any, **kwargs: Any) -> Optional[Any]:
         Returns:
             the output of :attr:`on_run_end` (often outputs collected from each step of the loop)
         """
+        if self.trainer is None:
+            raise MisconfigurationException(f"The {self.__class__.__name__} Loop hasn't been attached to any Trainer.")
```

Comment:
I think we should not enforce that if the loop does not need the trainer actually (e.g. if it is a custom one)

Reply:
Yes, let's remove this check.

Reply:
removed !